external/dhcpd/dhcpd_lwip.c

```diff
@@ -223,9 +226,24 @@ int dhcp_server_start(char *intf, dhcp_sta_joined dhcp_join_cb)
 	memcpy(data->intf, intf, strlen(intf) + 1);
 	data->fn = dhcp_join_cb;
 
-	int ret = ERROR;
 	g_dhcpd_term = 0;
-	ret = pthread_create(&g_dhcpd_tid, NULL, _dhcpd_join_handler, (void *)data);
+		/* Initialize the attribute variable */
+
+	if ((ret = pthread_attr_init(&attr)) != 0) {
+		ndbg("Failed to init attr\n");
+		free(data->intf);
+		free(data);
+		return ERROR;
+	}
+
+	sparam.sched_priority = 110;
+	if ((ret = pthread_attr_setschedparam(&attr, &sparam)) != 0) {
+		ndbg("Failed to set attr\n");
```

Comment:
Basically, most of kernel function including fs & nw from nuttx set errno when it failed internally.
So you'd better make a habit that add ret & errno in error log.